examples/md-flexible/Thermostat.h

```diff
@@ -12,120 +12,78 @@
-   */
-  void maxwellBoltzmannDistribution(autopas::Particle &p, const ThermostatFloatType factor) {
-    std::default_random_engine randomEngine(42);  // constant seed for repeatability
-    std::normal_distribution<double> normalDistribution{0, 1};
 
-    p.setV(autopas::utils::ArrayMath::addScalar(p.getV(), factor * normalDistribution(randomEngine)));
+/**
+ * Calculates temperature of system.
+ * @tparam AutoPasTemplate Type of AutoPas Object (no pointer)
+ * @tparam ParticlePropertiesLibraryTemplate Type of ParticlePropertiesLibrary Object (no pointer)
+ * @param autopas
+ * @param particlePropertiesLibrary
+ * @return Temperature of system.
+ */
+template<class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
+double calcTemperature(AutoPasTemplate &autopas, ParticlePropertiesLibraryTemplate &particlePropertiesLibrary) {
+  // kinetic energy times 2
+  double kineticEnergyMul2 = 0;
+#ifdef AUTOPAS_OPENMP
+#pragma omp parallel reduction(+ : kineticEnergyMul2)
+#endif
+  for (auto iter = autopas.begin(); iter.isValid(); ++iter) {
+    auto vel = iter->getV();
+    kineticEnergyMul2 +=
+        particlePropertiesLibrary.getMass(iter->getTypeId()) * autopas::utils::ArrayMath::dot(vel, vel);
   }
+  // AutoPas works always on 3 dimensions
+  constexpr unsigned int dimensions{3};
+  return kineticEnergyMul2 / (autopas.getNumberOfParticles() * dimensions);
+}
 
-  /**
-   * Initial temperature + deltaTemp * simstep / n_thermostat.
-   */
-  ThermostatFloatType _tInit;
-
-  /**
-   * Target temperature.
-   */
-  const ThermostatFloatType _tTarget;
-
-  /**
-   * Temperature difference per thermostat application until t_target is reached.
-   */
-  const ThermostatFloatType _deltaTemp;
-
-  /**
-   * ParticlePropertiesLibrary to access Mass of Particles
-   */
-  ParticlePropertiesLibraryTemplate _particlePropertiesLibrary;
-};
-template <class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
-void Thermostat<AutoPasTemplate, ParticlePropertiesLibraryTemplate>::addBrownianMotion(AutoPasTemplate &autopas,
-                                                                                       bool useCurrentTemp) {
+/**
+ * Adds brownian motion to the given system.
+ *
+ * If useCurrentTemp is set to true the factor of the brownian motion is calculated per particle based on its mass and
+ * the system's temperature. Otherwise a constant factor of 0.1 is used.
+ * Set this to false if the system is initialized without velocities.
+ *
+ * @tparam AutoPasTemplate Type of AutoPas Object (no pointer)
+ * @tparam ParticlePropertiesLibraryTemplate Type of ParticlePropertiesLibrary Object (no pointer)
+ * @param autopas
+ * @param particlePropertiesLibrary
+ * @param useCurrentTemp
+ */
+template<class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
+void addBrownianMotion(AutoPasTemplate &autopas,
+                       ParticlePropertiesLibraryTemplate &particlePropertiesLibrary,
+                       bool useCurrentTemp) {
```

Comment:
don't use the current temperature, it will break everything.
you might, however, want a prescribed temperature as an input.